components/Mediator.vue

```diff
@@ -552,7 +556,68 @@ function updateHandlerWindow(handlerWindow) {
 
 async function getWebAppManifest(host) {
   try {
-    const response = await axios.get('/manifest', {params: {host}});
+    const url = `https://${host}/manifest.json`;
+    if(!cacheStorage && typeof caches !== 'undefined') {
+      cacheStorage = await caches.open(CACHE_NAME);
+    }
+    if(!cacheStorage) {
+      // no cache storage/API available, fetch directly
+      return getWebAppManifestFromUrl(url);
+    }
+
+    // try to get cached response
+    let response = await cacheStorage.match(url);
+    if(response && response.ok) {
+      const expires = new Date(response.headers.get('expires'));
+      const now = new Date();
+      if(expires >= now) {
+        // return cached response
+        return response.json();
+      }
+      // remove expired response from cache
+      await cacheStorage.delete(url);
+    }
+
+    try {
+      // fetch live response
+      response = await fetch(url);
+
+      // build a cached response that will expire based on local config
+      const headers = new Headers(response.headers);
+      const expires = new Date(Date.now() + CACHE_TTL);
+      headers.set('expires', expires.toUTCString());
+      response = new Response(response.body, {
+        status: response.status,
+        statusText: response.statusText,
+        headers
+      });
+      await cacheStorage.put(url, response);
+    } catch(e) {
+      if(e.name === 'QuotaExceededError') {
+        // clear the whole cache
+        caches.delete(CACHE_NAME);
+        cacheStorage = null;
+        // use fetched response
+        return response.json();
+      }
+
+      // failed to fetch
+      console.error(e);
```

Comment:
I think the common 404 path comes through here?  Could this be a bit more friendly than an error?  At least some short text saying why this is logged.  And in the 404 case, and maybe others, perhaps this is more of a warning than an error?

Reply:
I filed an issue to experiment with what to do here. See: #90.